hazelcast/src/main/java/com/hazelcast/nio/tcp/BindHandler.java

```diff
@@ -88,6 +89,9 @@ private synchronized boolean bind(TcpIpConnection connection, ExtendedBindMessag
         // is created from an outbound port (eg 192.168.1.1:54003 --> 192.168.1.2:5701), but
         // in 192.168.1.2:5701's connectionsMap the connection must be registered with
         // key 192.168.1.1:5701.
+        assert (tcpIpEndpointManager.getEndpointQualifier() != EndpointQualifier.MEMBER
```

Comment:
Ensures that a `BindHandler` constructed in the context of a `MEMBER` endpoint manager will handle bind requests for connections with `MEMBER` connection type already set. Connection type of other connections can be set later (eg client connections are assigned their type upon client authentication).